modules/shared/src/main/scala/io/constellationnetwork/env/JarSignature.scala

```diff
@@ -14,8 +14,11 @@ import fs2.{Chunk, Stream}
 object JarSignature {
 
   def jarHash[F[_]: Async: Files]: F[Hash] = {
-    val jarPath = JarSignature.getClass.getProtectionDomain.getCodeSource.getLocation.toURI
-    digestOf(Files[F].readAll(Path(jarPath.getPath)))
+    val jarPath = Path(JarSignature.getClass.getProtectionDomain.getCodeSource.getLocation.toURI.getPath)
+    if (jarPath.extName === ".jar")
+      digestOf(Files[F].readAll(jarPath))
+    else
+      Hash("No jar found").pure
```

Comment:
Hmm, hash shouldn't be constructed like that. Either there should be a fallback to all zeros or sth like that, or imo preferable way would be to return option/either from this function, if we explicitly know it may fail

Reply:
I didn't want to complicate the code for a pathological case that only happens when running locally, but I'll change it to Option.